x/subscription/keeper/subscription_test.go

```diff
@@ -242,253 +168,210 @@ func TestCreateSubscription(t *testing.T) {
+	_, err = ts.TxSubscriptionBuy(sub1Addr, sub1Addr, plan.Index, 1)
 	require.NotNil(t, err)
 	require.Equal(t, uint64(12), sub.DurationLeft)
 	require.Equal(t, uint64(9), sub.DurationTotal)
 
 	// get the subscription's plan and make sure it uses the old plan
-	subPlan, found = ts.keepers.Plans.FindPlan(ts.ctx, sub.PlanIndex, sub.PlanBlock)
+	plan, found = ts.findPlan(sub.PlanIndex, sub.PlanBlock)
 	require.True(t, found)
-	require.Equal(t, oldPlanCuPerEpoch, subPlan.PlanPolicy.EpochCuLimit)
+	require.Equal(t, cuPerEpoch, plan.PlanPolicy.EpochCuLimit)
 
 	// delete the plan, and try to renew the subscription again
-	err = ts.keepers.Plans.DelPlan(ts.ctx, ts.plans[0].Index)
+	err = ts.Keepers.Plans.DelPlan(ts.Ctx, plan.Index)
 	require.Nil(t, err)
-	ts._ctx = keepertest.AdvanceEpoch(ts._ctx, ts.keepers)
-	ts.ctx = sdk.UnwrapSDKContext(ts._ctx)
+
+	ts.AdvanceEpoch()
 
 	// fast-forward another month, renewal should fail
-	ts.advanceMonths(timestamp, 1)
-	_, found = keeper.GetSubscription(ts.ctx, creator)
+	ts.AdvanceMonths(1).AdvanceEpoch()
+	_, found = ts.getSubscription(sub1Addr)
 	require.True(t, found)
-	err = keeper.CreateSubscription(ts.ctx, creator, creator, ts.plans[0].Index, 10)
+	_, err = ts.TxSubscriptionBuy(sub1Addr, sub1Addr, plan.Index, 10)
 	require.NotNil(t, err)
 }
 
 func TestSubscriptionAdminProject(t *testing.T) {
-	ts := setupTestStruct(t, 1)
-	keeper := ts.keepers.Subscription
+	ts := newTester(t)
+	ts.SetupAccounts(1, 0, 0) // 1 sub, 0 adm, 0 dev
 
-	account := common.CreateNewAccount(ts._ctx, *ts.keepers, 10000)
-	creator := account.Addr.String()
+	_, sub1Addr := ts.Account("sub1")
+	plan := ts.Plan("mock")
 
-	err := keeper.CreateSubscription(ts.ctx, creator, creator, ts.plans[0].Index, 1)
+	_, err := ts.TxSubscriptionBuy(sub1Addr, sub1Addr, plan.Index, 1)
 	require.Nil(t, err)
 
-	block := uint64(ts.ctx.BlockHeight())
-
 	// a newly created subscription is expected to have one default project,
 	// with the subscription address as its developer key
-	_, err = ts.keepers.Projects.GetProjectDeveloperData(ts.ctx, creator, block)
+	_, err = ts.Keepers.Projects.GetProjectDeveloperData(ts.Ctx, sub1Addr, ts.BlockHeight())
 	require.Nil(t, err)
 }
 
 func TestMonthlyRechargeCU(t *testing.T) {
-	ts := setupTestStruct(t, 1)
-	keeper := ts.keepers.Subscription
-	projectKeeper := ts.keepers.Projects
+	ts := newTester(t)
+	ts.SetupAccounts(1, 1, 1) // 1 sub, 1 adm, 1 dev
 
-	account := common.CreateNewAccount(ts._ctx, *ts.keepers, 10000)
-	anotherAccount := common.CreateNewAccount(ts._ctx, *ts.keepers, 10000)
-	creator := account.Addr.String()
+	_, sub1Addr := ts.Account("sub1")
+	_, adm1Addr := ts.Account("adm1")
+	_, dev1Addr := ts.Account("dev1")
+	plan := ts.Plan("mock")
 
-	blocksToSave, err := ts.keepers.Epochstorage.BlocksToSave(ts.ctx, uint64(ts.ctx.BlockHeight()))
+	_, err := ts.TxSubscriptionBuy(sub1Addr, sub1Addr, plan.Index, 3)
 	require.Nil(t, err)
 
-	// fill memory
-	for uint64(ts.ctx.BlockHeight()) < blocksToSave {
```

Comment:
is this really not needed for the test?

Reply:
the tests pass with and without it. 
I actually do not understand _why_ it was needed in the first place.
the `NewTester()` function kick-starts the blocks with a default AdvanceEpoch(). 
so the question is: is this really needed for any tests?

Reply:
I'm not sure either. It's @Yaroms code, we can ask him. Maybe something is wrong with the test